Dodge.js

```diff
@@ -1,178 +1,201 @@
 import { StatusBar } from 'expo-status-bar';
+import React, { Component, PureComponent } from 'react';
 import { StyleSheet, Text, View, Dimensions, Button, Image } from 'react-native';
-import { Knight, Circle, Brick, Grenade, MoveKnight, Shrapnel, counter} from "./Entities.js";
+import {
+  Knight, Circle, Brick, Grenade, MoveKnight,
+  Shrapnel, Water
+} from "./Entities.js";
 import { GameEngine } from "react-native-game-engine";
+import Matter from "matter-js";
 import AxisPad from 'react-native-axis-pad';
 ////this looks interesting for the attacker placing/throwing projectiles:
 //https://reactnative.dev/docs/panresponder 
 //https://medium.com/yapsody-engineering/hooks-and-function-components-in-react-native-d4e667c90cda
-var ents = {}
+let score = 0
+export default function DodgeBallGame({ navigation }) {
+  return (
+      <DodgeBall nav = {navigation}/>
+  )  
+}
 
-const generateUniqueId = require('generate-unique-id');
+class DodgeBall extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.ents = {}
+    this.counter = 0
+    this.nav = props.nav
+    const generateUniqueId = require('generate-unique-id');
```

Comment:
I'm not sure where you now use this :P

But let me give you some context: Like a counter, whatever algorithm this library is using needs some state to make sure the IDs are unique. That's why we say
```
const generateUniqueId = require('generate-unique-id'); // import and initialize the state *once*
```
rather than
```
// just import, there is no state unless you make an instance of a class you imported
import {generateUniqueId} from 'generate-unique-id'  
```

Since you want all IDs within a GameEngine to be unique, you need to store this state somewhere "high up". So probably you want to save it as:
```
this.generateUniqueId = require('generate-unique-id');
```
And then each Entity constructor can call (`this.id = this.parent.generateUniqueId()`)
[Or if the AddEntity function manages IDs, `entity.id = this.generateUniqueId(); this.ents[entity.id] = entity`]